src/extra/AutoUI/index.tsx

```diff
@@ -280,131 +282,134 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
+												onSearch={(term) => (
+													<FocusSearch
+														searchTerm={term}
+														filtered={filtered}
+														selected={selected}
+														setSelected={setSelected}
+														autouiContext={autouiContext}
+														model={model}
+														hasUpdateActions={hasUpdateActions}
+													/>
+												)}
 											/>
 										)}
-										<Update
-											model={model}
-											selected={selected}
-											autouiContext={autouiContext}
-											hasOngoingAction={false}
-											onActionTriggered={onActionTriggered}
-										/>
+									</Box>
+									{data.length > 0 && (
 										<HeaderGrid>
-											<LensSelection
-												lenses={lenses}
-												lens={lens}
-												setLens={(lens) => {
-													setLens(lens);
-													setToLocalStorage(
-														`${model.resource}__view_lens`,
-														lens.slug,
-													);
-												}}
+											{!!sdk?.tags && (
+												<Tags
+													autouiContext={autouiContext}
+													selected={selected}
+													changeTags={changeTags}
+												/>
+											)}
+											<Update
+												model={model}
+												selected={selected}
+												autouiContext={autouiContext}
+												hasOngoingAction={false}
+												onActionTriggered={onActionTriggered}
 											/>
+											<HeaderGrid>
+												<LensSelection
+													lenses={lenses}
+													lens={lens}
+													setLens={(lens) => {
+														setLens(lens);
+														setToLocalStorage(
+															`${model.resource}__view_lens`,
+															lens.slug,
+														);
+													}}
+												/>
+											</HeaderGrid>
 										</HeaderGrid>
-									</HeaderGrid>
-								)}
-							</HeaderGrid>
-							<Filters
-								renderMode={'summary'}
-								schema={model.schema}
-								filters={filters}
+									)}
+								</HeaderGrid>
+								<Filters
+									renderMode={'summary'}
+									schema={model.schema}
+									filters={filters}
+									autouiContext={autouiContext}
+									changeFilters={setFilters}
+								/>
+							</Box>
+							{data.length === 0 && (
+								<NoRecordsFoundArrow>
+									{t(`no_data.no_resource_data`, {
+										resource: t(`resource.item_plural`).toLowerCase(),
+									})}
+									<br />
+									{t('questions.how_about_adding_one')}
+								</NoRecordsFoundArrow>
+							)}
+						</>
+					)}
+					{!Array.isArray(data) && (
+						<HeaderGrid>
+							<LensSelection
+								lenses={lenses}
+								lens={lens}
+								setLens={(lens) => {
+									setLens(lens);
+									setToLocalStorage(`${model.resource}__view_lens`, lens.slug);
+								}}
+							/>
+						</HeaderGrid>
+					)}
+
+					{lens &&
+						data &&
+						(!Array.isArray(data) ||
+							(Array.isArray(data) && data.length > 0)) && (
+							<lens.data.renderer
+								flex={1}
```

Comment:
I think we also need to be wiring this in to the renderers, right?
But I guess that's not a blocker atm.

Reply:
Are you referring to `entity.tsx`'s and `table.tsx`'s renderer props? Given that the two work fine at the moment without the flex, I figured it was unnecessary to do and that we would instead wire it to any lenses that don't work fine without it (i.e. the device map lens)

Reply:
I have a feeling we might regret this later. I was going to pass for now, but on the other hand, someone could see that this `flex={1}` is then technically unused and remove it :thinking: